### PR TITLE
Don't warn about bsSize when it's not specified

### PR DESCRIPTION
**Pull request by taion**
Fixes a bug with #624

### CODE REVIEW

**Comment by taion** [comment]
Didn't realize `shouldNotWarn` was redundant. Got rid of it in revised commit.

**Comment by dozoisch** [comment]
looks good to me!

File: test/helpers.js [hunk 1]
```diff
@@ -1,9 +1,5 @@
-function shouldWarn(about) {
+export function shouldWarn(about) {
   console.warn.called.should.be.true;
   console.warn.calledWithMatch(about).should.be.true;
   console.warn.reset();
 }
-
-export default {
```
Comment (AlexKVal) [inline test/helpers.js]: That's my fault.
At some middle point of review process there was proposed version with additional exported function.
I've moved that function out, but forgot to simplify this.
:+1:

File: test/FormGroupSpec.js [hunk 2]
```diff
@@ -46,6 +46,15 @@ describe('FormGroup', function() {
     assert.ok(ReactTestUtils.findRenderedDOMComponentWithClass(instance, 'form-group-lg'));
   });
 
+  // This test case must come first, since the error only gets logged once.
```
Comment (AlexKVal) [inline test/FormGroupSpec.js]: It is just my curiosity.
I had placed this test after next one and then placed it into the end of file
and in both cases all tests are green.

What did you mean by this comment ?

Comment (taion) [inline test/FormGroupSpec.js]: Per https://github.com/facebook/react/blob/a56d0d6dc58a973d103bfae7e7ef7c04226209c7/src/classic/element/ReactElementValidator.js#L273 each prop type warning only gets logged once. Therefore, if I put this test case after the next one, no warning would be thrown even if my code didn't work.

Comment (AlexKVal) [inline test/FormGroupSpec.js]: Wow. I should be cautious now in my tests.
Thank you! :cherries:

Comment (mtscout6) [inline test/FormGroupSpec.js]: I really wish React had a way to turn that off, @aabenoja got pinned down on that for a while when working on the `ButtonInput` PR.

Comment (taion) [inline test/FormGroupSpec.js]: I agree. It was annoying enough that I was thinking of deleting this test case for a bit. "Doesn't throw a warning" isn't really an interesting or useful test in most cases.

**Pull request merged**